api/ruby/enterprise/change-domains-in-links.rb

```diff
@@ -0,0 +1,95 @@
+# Script to update the domain name for links in issue & pr comments.
+require 'octokit'
+
+## Check for environment variables
+begin
+  access_token = ENV.fetch("GITHUB_TOKEN")
+  hostname = ENV.fetch("GITHUB_HOSTNAME")
+rescue KeyError
+  puts
+  puts "To run this script, please set the following environment variables:"
+  puts "- GITHUB_TOKEN: A valid access token"
+  puts "- GITHUB_HOSTNAME: A valid GitHub Enterprise hostname"
+  exit 1
+end
+
+# Set up Octokit
+Octokit.configure do |kit|
+  kit.api_endpoint = "#{hostname}/api/v3"
+  kit.access_token = access_token
+  kit.auto_paginate = true
+end
+
+unless ARGV.length == 2
+  puts "Specify domain names to change using the following format:"
+  puts "- change-domains.rb old_domain new_domain"
+  exit 1
+end
+
+# Extract links to attached files using regexp
+# Looks for the raw markdown formatted image link formatted like this:
+# [image description](https://media.octodemo.com/user/267/files/e014c3e4-889c-11e6-8637-1f16c810cfe3)
+# example pattern = /\[[^\]]*\]\((https:\/\/[a-z]+.octodemo.com[^\)]*\/files\/[^\)]*)\)/
+old_domain    = ARGV[0]
+new_domain    = ARGV[1]
+media_pattern = /\[[^\]]*\]\((https:\/\/[a-z]+.#{old_domain}[^\)]*\/user\/\d*\/files\/[^\)]*)\)/
```

Comment:
Thanks for testing that. Should we then make sure to construct all new links as `https://[new_hostname]/storage/user/...`, then let the instance forward to the subdomain as needed?